accord-core/src/main/java/accord/coordinate/FetchData.java

```diff
@@ -149,11 +150,15 @@ private static Object fetchWithIncompleteRoute(Known fetch, Node node, TxnId txn
 
     public static Object fetch(Known fetch, Node node, TxnId txnId, FullRoute<?> route, @Nullable Timestamp executeAt, BiConsumer<Known, Throwable> callback)
     {
-        Ranges ranges = node.topology().localRangesForEpochs(txnId.epoch(), fetch.fetchEpoch(txnId, executeAt));
-        return fetchInternal(ranges, fetch, node, txnId, route.sliceStrict(ranges), executeAt, callback);
+        AsyncChain<Object> chain = node.awaitEpoch(executeAt).map(ignore -> {
```

Comment:
this is prob needed in other methods... its really hard to tell... this failed for me, so just changed this single code path

Reply:
Something seems off because on your branch this method isn't called, the method looks like it should really be private, and the other methods either automatically go and get the epoch already, or have an invariant asserting it is available.

Reply:
> this method 

can you define `this`?  `node.awaitEpoch` is new to this patch... and this diff shows it as used

Reply:
This `fetch` method is what I am talking about not `awaitEpoch` which is if you remove this `fetch` method also ends up being unused.

Reply:
This looks like a rebase issue, the method should have `BiConsumer<? super Known, Throwable> callback` but that got lost in the rebase and became `BiConsumer<Known, Throwable> callback` which made the partial route one called 100% of the time... once I fixed the signature then this is called from SPL